presto-spi/src/main/java/com/facebook/presto/spi/block/VariableWidthBlock.java

```diff
@@ -132,7 +132,7 @@ public void retainedBytesForEachPart(BiConsumer<Object, Long> consumer)
     @Override
     public Block copyPositions(int[] positions, int offset, int length)
     {
-        checkValidPositions(positions, offset, length, positionCount);
+        checkValidPositionsArray(positions, offset, length);
```

Comment:
missing `checkValidPosition(positionCount, position);` below?

Reply:
same, every position is validated